host.go

```diff
@@ -305,7 +306,44 @@ func (h *gpbftHost) ReceiveDecision(decision *gpbft.Justification) time.Time {
 		return time.Now().Add(h.manifest.ECDelay)
 	}
 
-	return ts.Timestamp().Add(h.manifest.ECDelay)
+	if !decision.Vote.Value.IsOnlyBase() {
+		// we decided on something new, use just the ECDelay
+		return ts.Timestamp().Add(h.manifest.ECDelay)
+	}
+
+	// we decided on base, calculate how much we should back off
+	// all of this should go into manifest but I think Alfonso dislikes me already :P
+	const (
+		minBackoff = 2.
+		maxBackoff = 3. * 2. * 60. // 3h with 30s ECDelay
+	)
+	// the backoff is defined in multiples of ECDelay starting at the last finalized tipset
+	// each additional base decision beyond that will incurr the maxBackoff
+	var backoffTable = []float64{2, 3, 5, 9, 17, 33, 65, 129, 257, 513} // 2^i+1 backoff, table for more flexibility
+	//TODO move all the above to manifest
+
+	attempts := 0
+	var backoffMultipler float64
+	for instance := decision.Vote.Instance - 1; instance > h.manifest.InitialInstance; instance-- {
+		cert, err := h.client.certStore.Get(h.runningCtx, instance)
```

Comment:
Is cert store guaranteed to have the complete set of certificates? What the unhappy path would look like if not? Errors here could halt the entire EC.

Reply:
Yes, certstore is guaranteed to have all certs.